### DISCUSSION TITLE
Message Received Not Working

### DISCUSSION THREAD

**Discussion by Pluppers** [discord-jda/JDA#2152] [Questions and Help]
My MessageReceived is not working for no reason at all. Is there any way that Messagereceived does not work? I think I initialized properly and other events work, just not message received?

**Comment by MinnDevelopment** [comment]
Take a look at this https://jda.wiki/using-jda/troubleshooting/#my-event-listener-code-is-not-executed

**Reply by Pluppers** [reply]
This is super helpful, however, another event is working an I don't know why its just the Message event that is not working.

**Reply by Splayfery** [reply]
Have you registered your event in the main class? Maybe try to use @Override at the top of the method

**Comment by yusuf1746** [comment]
is there a red line under it or is it just not working?

**Reply by yusuf1746** [reply]
if no try this
```
@Override
    public void onMessageReceived(MessageReceivedEvent event) {
		
		event.getChannel().sendMessage("hi").queue();
		
    }
```
